### DISCUSSION TITLE
Can't get the Comments feature to work

### DISCUSSION THREAD

**Discussion by thomasmattheussen** [vercel/community#1441] [Help] [answered]
I can't get the Comments UI toolbar to show up on my preview deployment URL. I enabled the feature in the settings and deployed a new preview, but it doesn't seem to work.

Does this have anything to do with the fact that I'm deploying a private repository through the Vercel CLI?

I also don't understand how it should work in the first place though. I'm guessing there's some piece of javascript that needs to be included on the page, but I have no idea where that should come from?

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @thomasmattheussen. The [Comments](https://vercel.com/docs/concepts/deployments/comments#) feature is still in [beta](https://vercel.com/docs/concepts/release-phases#beta) and doesn't work with CLI deployments at the moment. There is some setup magic that happens behind the scenes to make this feature work, so it is only available with [git integration](https://vercel.com/docs/concepts/git#) deployments for now.

**Reply by thomasmattheussen** [reply]
Ok, thanks for the info! 
Might be worth it to add this to the documentation? Or did I miss it somewhere?

**Reply by amyegan** [reply]
It seems to be more implied than explicitly stated in the documentation. Good idea to make it more clear. I'm sharing your feedback with the documentation team. Thanks!

**Reply by gdborton** [reply]
Wanted to add a tiny bit of additional context, but CLI deploy support is coming soon!

When we've landed support, there'll be nothing you need to do beyond enabling the feature. We make the magic happen at build/request time.

**Reply by thomasmattheussen** [reply]
Sounds great! 😊

**Reply by martijndierckx** [reply]
@gdborton do you have an estimated timeline on this update?

**Reply by amyegan** [reply]
Preview Comments work with both the Git integration and the CLI now! Please let us know if you have any trouble with this going forward 🙇

**Reply by thomasmattheussen** [reply]
Hi @amyegan , while it was working great for us the past few weeks, something seems to have been changed. 
We can now no longer use the comments feature on our **CLI deployed** preview deployments. When clicking the "_Login to comment_" button at the bottom, everyone gets the error: 

> "The deployment did not originate from Git; therefore, it cannot be used."

**Reply by thomasmattheussen** [reply]
To be clear: we don't have any git integration linked in our project (because we use a self managed gitlab instance)
This seemed to work fine though, but something changed...

**Reply by amyegan** [reply]
Thanks for letting me know, Thomas. I shared this info with the team and I'll let you know when I learn more
